tool/tsh/putty_config_windows.go

```diff
@@ -0,0 +1,384 @@
+	if runtime.GOOS != constants.WindowsOS {
+		return trace.NotImplemented("PuTTY config is only implemented on Windows")
+	}
+
+	tc, err := makeClient(cf, true)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	// connect to proxy to fetch cluster info
+	proxyClient, err := tc.ConnectToProxy(cf.Context)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	defer proxyClient.Close()
+
+	// parse out proxy details
+	proxyHost, _, err := net.SplitHostPort(tc.Config.SSHProxyAddr)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	// get root cluster name and set keypaths
+	rootClusterName, err := proxyClient.RootClusterName(cf.Context)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	keysDir := profile.FullProfilePath(tc.Config.KeysDir)
+	ppkFilePath := keypaths.PPKFilePath(keysDir, proxyHost, tc.Config.Username)
+	certificateFilePath := keypaths.SSHCertPath(keysDir, proxyHost, tc.Config.Username, rootClusterName)
+
+	// process leaf clusters if --leaf flag is passed
+	if cf.LeafClusterName != "" {
+		leafClusters, err := proxyClient.GetLeafClusters(cf.Context)
+		if err != nil {
+			return trace.Wrap(err)
+		}
+		// check whether we know about the leaf cluster named, error if not
+		leafClustersNames := make([]string, 0, len(leafClusters))
+		for _, leafCluster := range leafClusters {
+			leafClustersNames = append(leafClustersNames, leafCluster.GetName())
+		}
+		if !slices.Contains(leafClustersNames, cf.LeafClusterName) {
+			return trace.BadParameter("Cannot find registered leaf cluster %q. Use the leaf cluster name as it appears in the output of `tsh clusters`.", cf.LeafClusterName)
+		}
+		certificateFilePath = keypaths.SSHCertPath(keysDir, proxyHost, tc.Config.Username, cf.LeafClusterName)
+	}
+
+	// get all CAs for the cluster (including trusted clusters)
+	var cas []types.CertAuthority
+	err = tc.WithRootClusterClient(cf.Context, func(clt auth.ClientI) error {
+		cas, err = clt.GetCertAuthorities(cf.Context, types.HostCA, false /* exportSecrets */)
+		if err != nil {
+			return trace.Wrap(err)
+		}
+		return nil
+	})
+	// iterate over CAs
+	hostCAPublicKeys := make(map[string][]string)
+	for _, ca := range cas {
+		var publicKeys []string
+		// if this is either the root or requested leaf cluster, process it
+		if ca.GetName() == rootClusterName || ca.GetName() == cf.LeafClusterName {
+			for i, key := range ca.GetTrustedSSHKeyPairs() {
+				log.Debugf("%v CA [%v]: %v", ca.GetName(), i, key)
+				kh, err := sshutils.MarshalKnownHost(sshutils.KnownHost{
+					Hostname:      ca.GetClusterName(),
+					AuthorizedKey: key.PublicKey,
+				})
+				if err != nil {
+					return trace.Wrap(err)
+				}
+				_, _, hostCABytes, _, _, err := ssh.ParseKnownHosts([]byte(kh))
+				if err != nil {
+					return trace.Wrap(err)
+				}
+				publicKeys = append(publicKeys, strings.TrimPrefix(strings.TrimSpace(string(ssh.MarshalAuthorizedKey(hostCABytes))), constants.SSHRSAType+" "))
+				hostCAPublicKeys[ca.GetName()] = publicKeys
```

Comment:
Can we reduce this to `hostCAPublicKeys[ca.GetName()] = append(hostCAPubilcKeys[ca.GetName(), ..`?

I had to read through the logic several times to understand what was going on here.